src/main/java/core/basesyntax/FileWork.java

```diff
@@ -7,34 +7,34 @@
 
 public class FileWork {
     private static final String CHECK_LETTER = "w";
+    private static final String CHECK_REGEX = "[!\\.,?]";
 
     public String[] readFromFile(String fileName) {
-        String textFromFile = "";
+        StringBuilder textFromFile = new StringBuilder();
         FileReader fileReader;
         try {
             fileReader = new FileReader(fileName);
-        } catch (IOException e) {
-            throw new RuntimeException("File not found", e);
-        }
-        try {
             BufferedReader bufferedReader = new BufferedReader(fileReader);
-            while (bufferedReader.ready()) {
-                String word = bufferedReader.readLine();
-                String[] words = word.split(" ");
-                for (String string : words) {
-                    if (string.toLowerCase().startsWith(CHECK_LETTER)) {
-                        textFromFile += string.toLowerCase().replaceAll("[!\\.,?]", "") + " ";
-                    }
+            String line = bufferedReader.readLine();
+            StringBuilder stringBuilder = new StringBuilder();
+            while (line != null) {
+                stringBuilder.append(line).append(" ");
+                line = bufferedReader.readLine();
+            }
```

Comment:
you are making the same job twice - gather data together to split it afterwards. lets process data when we receive it